src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -3058,7 +3058,7 @@ public override void Visit(FirstNameNode node)
                 }
 
                 // Any connectedDataSourceInfo or option set or view needs to be accessed asynchronously to allow data to be loaded.
-                if (lookupInfo.Data is IExternalTabularDataSource || lookupInfo.Kind == BindKind.OptionSet || lookupInfo.Kind == BindKind.View)
+                if (lookupInfo.Data is IExternalTabularDataSource || lookupInfo.Kind == BindKind.OptionSet || lookupInfo.Kind == BindKind.View || lookupInfo.IsAsync)
```

Comment:
>lookupInfo.Data is IExternalTabularDataSource || lookupInfo.Kind == BindKind.OptionSet || lookupInfo.Kind == BindKind.View  [](http://example.com/codeflow?start=20&length=123)

Eventually, these checks should get removed from the binder and we should just use the IsAsync flag. 

Idea to consider: 
As an intermediate step, we could move this check into the NameLookupInfo ctor and set IsAync == true if these other flags are set. 
That lets us simplify the binder.cs now; and then remove that check altogether when PAclient is eventually updated.